pipelines/monitor_forecast_trigger.py

```diff
@@ -0,0 +1,341 @@
+            logger.info("Action trigger condition met; sending email.")
+            subject = f"Anticipatory action Fiji: Cyclone {cyclone_name} action trigger ACTIVATED"
+            body = render_template(
+                "action.html",
+                {
+                    "cyclone_name": cyclone_name,
+                    "forecast_display_str": forecast_display_str,
+                },
+            )
+            create_and_send_campaign(
+                subject=subject,
+                name=f"{email_base_name}_action",
+                list_ids=TRIGGER_LIST_IDS,
+                body=body,
+            )
+        else:
+            logger.info("Trigger conditions not met; no emails sent.")
+
+    # Calculate perturbed tracks exposure at adm0 level
+    (
+        df_exp_shift,
+        gdf_shift_buffers,
+        gdf_shift_tracks,
+    ) = calculate_shifted_exposures(
+        gdf_readiness, da_wp_clip, disable_tqdm=DISABLE_TQDM
+    )
+    df_exp_shift = df_exp_shift.sort_values(
+        [f"exp_{x}" for x in [64, 50, 34]], ascending=False
+    )
+    worst_row = df_exp_shift.iloc[0].copy()
+    worst_row["level"] = "worst"
+    best_row = df_exp_shift.iloc[-1].copy()
+    best_row["level"] = "best"
+    logger.info(
+        "Best case exposure:\n%s",
+        best_row.to_frame().T.to_string(index=False),
+    )
+    logger.info(
+        "Worst case exposure:\n%s",
+        worst_row.to_frame().T.to_string(index=False),
+    )
+    df_stats = load_historical_stats()
+
+    # Produce thermometer plot
+    fig_thermometer, ax = plot_thermometer(
+        main_value=readiness_exp,
+        low_bound=best_row["exp_64"],
+        high_bound=worst_row["exp_64"],
+        df_stats=df_stats,
+        cyclone_name=cyclone_name,
+        forecast_display_str=forecast_display_str,
+    )
+    img_base64_thermometer = fig_to_base64(fig_thermometer)
+
+    # Calculate exposure at adm3 level for most likely
+    logger.info("Calculating ADM3 level exposure for most likely track.")
+    df_exp_adm3_mostlikely = calculate_multi_adm_exposure(
+        gdf_buffers_readiness, da_wp_clip, adm3, disable_tqdm=DISABLE_TQDM
+    )
+
+    # Save adm3 exposure CSV file for attachment to email
+    df_adm3_out = df_exp_adm3_mostlikely.pivot(
+        columns="buffer_speed", index="ADM3_PCODE", values="pop_exposed"
+    )
+    df_adm3_out = df_adm3_out.rename(
+        columns={x: f"exp_{x}_knot" for x in df_adm3_out.columns}
+    )
+    df_adm3_out = df_adm3_out.reset_index()
+    df_adm3_out.columns.name = None
+    cols = [
+        "ADM1_PCODE",
+        "ADM1_EN",
+        "ADM2_PCODE",
+        "ADM2_EN",
+        "ADM3_PCODE",
+        "ADM3_EN",
+    ]
+    df_adm3_out = adm3[cols].merge(df_adm3_out)
+    df_adm3_out = df_adm3_out.sort_values(
+        [f"exp_{x}_knot" for x in [64, 50, 34]], ascending=False
+    )
+    adm3_exp_filename = f"temp/{forecast_id}_adm3_exposure.csv"
+    if not os.path.exists("temp/"):
+        os.makedirs("temp/")
```

Comment:
just learned this one which is apparently safer and cleaner:

```
os.makedirs("temp/", exist_ok=True)  
```

no biggie though as the current version can only be a problem if multiple processes are running and can mess this step up depending on the order they get to it ("race conditions")